app/assets/javascripts/Components/assignment_summary_table.jsx

```diff
@@ -1,18 +1,27 @@
 import React from "react";
 import {render} from "react-dom";
-import {markingStateColumn} from "./Helpers/table_helpers";
+import {markingStateColumn, getMarkingStates} from "./Helpers/table_helpers";
 
 import ReactTable from "react-table";
 
 class AssignmentSummaryTable extends React.Component {
   constructor() {
     super();
+    const markingStates = {
```

Comment:
Eh, instead of duplicating the keys here, why don't you call `getMarkingStates` on an empty list? (Similar for the other constructor below.